CMakeLists.txt

```diff
@@ -19,7 +19,18 @@ set(HEADERS
 
 link_directories(${TANGO_PKG_LIBRARY_DIRS})
 
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+if(COMPILER_SUPPORTS_CXX11)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+elseif(COMPILER_SUPPORTS_CXX0X)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+else()
+    message("The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. You may not benefit from performance optimizations.")
+endif()	
+
 add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS})
```

Comment:
```suggestion
add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS})
set_target_properties(${PROJECT_NAME} PROPERTIES
                      CXX_STANDARD 11
                      CXX_EXTENSIONS NO)

```

Reply:
Using something like the above suggestion might be even more portable.
I've tested this on Debian wheezy and it compiles using `-std=c++0x` compiler option.

Reply:
We can also do it directly at the project level:
```
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_EXTENSIONS OFF)
```
We'll need to raise the minimum CMake required version too:
cmake_minimum_required(VERSION 3.1.3)

Reply:
What you propose is a good change in my opinion. I'm in favor of setting these flags at project level.

Reply:
> What you propose is a good change in my opinion. I'm in favor of setting these flags at project level.

@mliszcz, done. Thanks for your feedback.